src/lib/libscf_solver/hf.cc

```diff
@@ -405,12 +405,12 @@ void HF::integrals()
           #ifdef HAVE_JK_FACTORY
             //DGAS is adding to the ghetto, this Python -> C++ -> C -> C++ -> back to C is FUBAR
             boost::shared_ptr<Molecule> other_legacy = Process::environment.legacy_molecule();
-            Process::environment.set_legacy_wavefunction(molecule_);
+            //Process::environment.set_legacy_wavefunction(Process::environment.legacy_wavefunction());
+            Process::environment.set_legacy_molecule(other_legacy);
             if(options_.get_bool("SOSCF"))
                 jk_ = boost::shared_ptr<JK>(new GTFockJK(basisset_,2,false));
             else
                 jk_ = boost::shared_ptr<JK>(new GTFockJK(basisset_,2,true));
-            Process::environment.set_legacy_molecule(other_legacy);
           #else
```

Comment:
I am not quite sure I understand what you are trying to do here, the legacy molecule should not be passed to GTFock.

The intention of this code was to set the current HF molecule to `legacy_molecule` so that GTFock can scrape the data it needs and then revert to the old legacy molecule so that this temporary flip is transparent.